### PR TITLE
Add Ensembl endpoints

### PR DESCRIPTION
**Pull request by inodb**
Add Python scripts to download tables from Ensembl. Perform data
transformations and load data in Genome Nexus to serve over ensembl endpoints.
Importing data first in mongodb on init.

### cBioPortal Usages:
- We want to replace the current pfam service to use genome nexus instead.
- Display appropriate length of protein in mutation diagram based on chosen isoform override (this will be a new property in cbioportal/cbioportal)

### CODE REVIEW

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/EnsemblController.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController // shorthand for @Controller, @ResponseBody
+@CrossOrigin(origins="*") // allow all cross-domain requests
+@RequestMapping(value= "/")
+@Api(tags = "ensembl-controller", description = "Ensembl Controller")
```
Comment (onursumer) [inline annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/EnsemblController.java]: Also add `@PublicApi` annotation?

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/domain/EnsemblTranscript.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.univocity.parsers.annotations.Parsed;
+import com.univocity.parsers.annotations.Trim;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * Ensembl transcript
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (onursumer) [inline annotation/src/main/java/org/cbioportal/genome_nexus/annotation/domain/EnsemblTranscript.java]: Ideally we should add API related annotations like `@JsonInclude` and `@ApiModelProperty` in a mixin under web package, but we can also do that in another PR.

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/GenomeNexusAnnotation.java [hunk 3]
```diff
@@ -55,8 +61,14 @@
  */
 @SpringBootApplication // shorthand for @Configuration, @EnableAutoConfiguration, @ComponentScan
 @EnableSwagger2 // enable swagger2 documentation
-public class GenomeNexusAnnotation extends SpringBootServletInitializer
+public class GenomeNexusAnnotation extends SpringBootServletInitializer implements CommandLineRunner
```
Comment (onursumer) [inline annotation/src/main/java/org/cbioportal/genome_nexus/annotation/GenomeNexusAnnotation.java]: Would it be too difficult to implement the `CommandLineRunner` in a separate class? Something like `class GenomeNexusImporter implements CommandLineRunner`.

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/GenomeNexusAnnotation.java [hunk 4]
```diff
@@ -102,4 +114,85 @@ private ApiInfo apiInfo() {
...
     }
+
+    /**
+     * Export a resource embedded into a Jar file to the local file path.
+     *
+     * @param resourceName ie.: "/SmartLibrary.dll"
+     * @return The path to the exported resource
+     * @throws Exception
+     */
+    static public String ExportResource(String resourceName) throws Exception {
```
Comment (onursumer) [inline annotation/src/main/java/org/cbioportal/genome_nexus/annotation/GenomeNexusAnnotation.java]: It is against Java coding conventions to have a method name starting with an uppercase letter :) Can we rename it to `exportResource`?

Comment (inodb) [reply annotation/src/main/java/org/cbioportal/genome_nexus/annotation/GenomeNexusAnnotation.java]: @onursumer ah yes, this function was a copy+paste from SO. Guess this was some C++/C# dev: https://www.reddit.com/r/java/comments/3cu10n/do_any_of_you_start_static_methods_with_a_capital/

**Comment by inodb** [comment]
@onursumer ok should be good now

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/EnsemblController.java (hunk 1 above)
Comment (inodb) [reply annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/EnsemblController.java]: We decided to go for internal instead, because we're not 100% happy about the endpoints yet (specifically isoform override)

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/domain/EnsemblTranscript.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    private String transcriptId;
+
+    @Field(value="gene_stable_id")
+    private String geneId;
+
+    @Field(value="protein_stable_id")
+    private String proteinId;
+
+    @Field(value="protein_length")
+    private String proteinLength;
```
Comment (onursumer) [inline annotation/src/main/java/org/cbioportal/genome_nexus/annotation/domain/EnsemblTranscript.java]: Better to make this an `Integer`

File: annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/mixin/EnsemblTranscriptMixin.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    private String transcriptId;
+
+    @ApiModelProperty(value = "Ensembl gene id", required = true)
+    private String geneId;
+
+    @ApiModelProperty(value = "Ensembl protein id", required = true)
+    private String proteinId;
+
+    @ApiModelProperty(value = "Length of protein")
+    private String proteinLength;
```
Comment (onursumer) [inline annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/mixin/EnsemblTranscriptMixin.java]: This one needs to change to `Integer` too

**Comment by onursumer** [comment]
@inodb I think this is good to merge, except the type of `proteinLength`
